keywrap/pkcs11/keywrapper_pkcs11_test.go

```diff
@@ -0,0 +1,138 @@
+/*
+   Copyright The ocicrypt Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package pkcs11
+
+import (
+	"os"
+	"testing"
+
+	"github.com/containers/ocicrypt/config"
+	"github.com/containers/ocicrypt/crypto/pkcs11"
+	"github.com/containers/ocicrypt/utils/softhsm"
+)
+
+var (
+	SOFTHSM_SETUP = "../../scripts/softhsm_setup"
+)
+
+func getPkcs11Config(t *testing.T) pkcs11.Pkcs11Config {
+	// we need to provide a configuration file so that on the various distros
+	// the libsofthsm2.so will be found by searching directories
+	config := `module-directories:
+ - /usr/lib64/pkcs11/  # Fedora
+ - /usr/lib/softhsm/   # Ubuntu
+`
+	p11conf, err := pkcs11.ParsePkcs11ConfigFile([]byte(config))
+	if err != nil {
+		t.Fatal(err)
+	}
+	return p11conf
+}
+
+func createValidPkcs11Ccs(t *testing.T) ([]*config.CryptoConfig, error) {
+	// We always need the query attributes  'pin-value' and 'module-name'
+	// for SoftHSM2 the only other important attribute is 'object' (= the 'label')
+	pkcs11PubKeyUriStr, err := softhsm.RunSoftHSMSetup(SOFTHSM_SETUP)
+	if err != nil {
+		return nil, err
+	}
+	pubKeyPem, err := softhsm.RunSoftHSMGetPubkey(SOFTHSM_SETUP)
+	if err != nil {
+		return nil, err
+	}
+	pkcs11PrivKeyYaml := `pkcs11:
```

Comment:
Hmm, would it be cleaner for the tests if we used an anonymous struct and `yaml.Marshal`?

Reply:
The string is supposed to be a representation of file contents as one would find in the filesystem.

Reply:
yea that's fine then... I just have a pet peeve for including "indent sensitive" strings in non-indent based languages. But since it's only in one area, it seems reasonable.